### DISCUSSION TITLE
What is the difference between the `ExternalFn` and `Fn` expressions?

### DISCUSSION THREAD

**Discussion by CrowdHailer** [gleam-lang/gleam#1021] [Questions & support] [answered]
I've been working under the (i assume) incorrect assumption that the difference here relates to whether the functions are public or not. I've just discovered a public field so can't work out what ExternalFn vs Fn means.

https://github.com/gleam-lang/gleam/blob/main/src/erl.rs#L189-L201

**Comment by lpil** [comment, marked as answer]
`Fn` is `fn main() { do_stuff() }` while `ExternalFn` is `external fn main() -> Int = "module" "function"`.

Some doc comments could help here for sure
